Various 3D mode fixes again

ryotoko opened the pull request.

ryotoko committed:
Fix 3D mode skybox clipping problem

- Changes tile grid logic from 3x3 to 5x5 if 3D mode is enabled
- Vertex clip rejection y-axis test range also needed expanding

ryotoko committed:
Various 3D mode fixes again

Fixes skybox clipping (tidy up)

Implements new function that allows setting IOD at various points in the game. Fixes:
- File select menu
- Star select menu
- Cannon reticle screen

Fixes clipping related to new implementation of stereoscopic separation
- Screen transitions, menus, etc
- Peach ending scene

Adjusts values for new stereoscopic separation

ryotoko committed:
IOD function tidy up

IOD is now changed via RDP dlist

ryotoko committed:
Goddard background is now widescreen

-5x3 tiles fit perfectly in 2D
-3D mode has clipped tiles on both sides

mkst commented on include/PR/gbi.h:
```diff
@@ -4810,6 +4811,21 @@ typedef union {
 	_g->words.w0 = _SHIFTL(G_SPECIAL_2, 24, 8); \
 }
 
+#define gDPSetIod(pkt, iod) \
```
you're mixing tabs and spaces here

mkst commented on src/menu/star_select.c:
```diff
@@ -378,6 +378,7 @@ Gfx *geo_act_selector_strings(s16 callContext, UNUSED struct GraphNode *node) {
 #endif
     if (callContext == GEO_CONTEXT_RENDER) {
 #ifdef ENABLE_N3DS_3D_MODE
+        gDPSetIod(gDisplayListHead++, iodStarSelect);
```
did you try adding these AFTER the flush?

The flush results in a call to draw_triangles

mkst commented on src/pc/gfx/gfx_citro3d.c:
```diff
@@ -710,15 +744,14 @@ void gfx_citro3d_frame_draw_on(C3D_RenderTarget* target)
 {
 #ifdef ENABLE_N3DS_3D_MODE
-    Mtx_Identity(&projection);
+    Mtx_Identity(&projection); // this doesn't seem needed
```
I think this is to ensure if we drew something 3D then 2D we reset the projection matrix?

I think it's hard to tell, but if you set the matrix up to adjust x/y you'll find the PRESS START is offset on Goddard along with other 2D things like HUD.

mkst commented on include/PR/gbi.h:
```diff
@@ -117,6 +117,7 @@
 #define G_SPECIAL_1		0xd5
 #define G_SPECIAL_2		0xd4
 #define G_SPECIAL_3		0xd3
+#define G_SPECIAL_4		0xd2
```
might want an `#ifdef` around this, the first 3 SPECIALs are vanilla 👍

ryotoko committed:
3D mode improvements and cleanup

Set2D now has several modes that flag specific asset handling cases

Goddard screen reworked
- Background increased by 1 rile column (exact fit)
- 3D effect strengthened
- "Press start" separated from render_text_labels for individual effect
- Goddard hand returned to 3D

General cleanup

ryotoko committed:
Score menu crash in 3D mode and cleanup
